syncer/syncer.go

```diff
@@ -153,26 +161,33 @@ func (s *Syncer) IsSynced(ctx context.Context) bool {
 // Start starts the main sync loop that tries to sync data for every SyncInterval
 func (s *Syncer) Start(ctx context.Context) {
 	s.syncOnce.Do(func() {
-		if s.ticker.GetCurrentLayer().Uint32() <= 1 {
-			s.setSyncState(ctx, synced)
-		}
-		for {
-			select {
-			case <-s.shutdownCtx.Done():
-				s.logger.WithContext(ctx).Info("stopping sync to shutdown")
-				return
-			case <-s.syncTimer.C:
-				s.synchronize(ctx)
+		s.logger.WithContext(ctx).Info("Starting syncer loop")
+		s.eg.Go(func() error {
+			if s.ticker.GetCurrentLayer().Uint32() <= 1 {
+				s.setSyncState(ctx, synced)
 			}
-		}
+			for {
+				select {
+				case <-s.shutdownCtx.Done():
+					s.logger.WithContext(ctx).Info("stopping sync to shutdown")
+					return s.shutdownCtx.Err()
+				case <-s.syncTimer.C:
+					s.logger.WithContext(ctx).Debug("synchronize on tick")
+					s.synchronize(ctx)
+				case <-s.forceSyncCh:
+					s.logger.WithContext(ctx).Debug("force synchronize")
+					s.synchronize(ctx)
+				}
+			}
+		})
 	})
 }
 
 // ForceSync manually starts a sync process outside the main sync loop. If the node is already running a sync process,
 // ForceSync will be ignored.
 func (s *Syncer) ForceSync(ctx context.Context) {
 	s.logger.WithContext(ctx).Debug("executing ForceSync")
-	go s.synchronize(ctx)
+	s.forceSyncCh <- struct{}{}
```

Comment:
this needs to either select on shutdown ctx or drop a token if one already exists, otherwise it may hang if loop exited

Reply:
done. thanks. added test too